### DISCUSSION TITLE
Centralization measures

### DISCUSSION THREAD

**Discussion by marielouiseferguson** [networkx/networkx#7629] [Ideas]
Hello,

It would be great if centralization measures for the network as a whole could be added. There is currently support for degree/eigenvector centrality and closeness/betweenness calculations. It would be awesome if there was a function to also calculate the centralization of the network using these measures, like is done in R in iGraph.

--> https://r.igraph.org/reference/centralize.html#references

Thanks!

**Comment by dschult** [comment]
This sounds like a good idea, especially if it follows ideas or formulas from those two references in the igraph docs. 

The devil in the details here is that there are many special cases for what type of graph we are considering and that comes across most messily when trying to normalize the results. I like this idea of letting the user specify what to normalize with respect to, because it is hard to account for all the hidden assumptions in a general way. 

I'm not sure I understand the formula as written in those docs -- It looks like a sum of the variation in centrality scores. So if all the nodes have the same (possibly high) centrality value, the network's centrality score would be zero. Can you state how you think about these scores? Are we looking for the variation in node centrality as a measure of the centrality of a network? How can I think about this.

**Reply by marielouiseferguson** [reply]
This is paper the idea is drawn from, you can find the calculations for centralization here on page 16 for degree centrality and betweenness/closeness: https://bebr.ufl.edu/sites/default/files/Centrality%20in%20Social%20Networks.pdf

The sum of differences is divided by the maximum theoretical possible sum of differences for a graph with that many nodes. So close to 0 would indeed mean all points in the network are equal (great) and close to one would mean graph inequality.

My issue is that I see in your docs that degree centrality is normalized by (n-1). Whereas in the paper degree centralization (or the differences) are divided by (n-1)(n-2). So that means potentially I would have to multiply your measure by (n-1)/((n-1)(n-2)) i.e. divide by (n-2). This is the way I'm thinking about it anyway. But it would be great if this was incorporated in the package, even if just for simple graphs at first.

I found a gist here which has attempted it, but I don't think he took the networkx normalization into account: https://gist.github.com/aldous-rey/e6ee7b0e82e23a686d5440bf3987ee23

**Reply by dschult** [reply]
Can you briefly describe what you are trying to measure in a network that this measure would indicate?
The Freeman paper from 1979 doesn't seem to have much to follow it up. And it is difficult to find papers within the last 15 years that talk about network measures of centralization. What are we measuring and why?

And for the normalization, it looks like page 221 normalizes degree centrality by dividing by (n-1).  That is, not by (n-1)*(n-2). And degree centrality should be normalized in this way because the maximum degree is (n-1).

**Reply by marielouiseferguson** [reply]
I am trying to use it as a measure of interconnectedness in a graph as a whole/whether the network is dominated by one node with very high degree or equally spread out. If centralization is close to 0, then the graph is equally connected between all the nodes. If it is closer to one, then one node has all the connections and is very central while the others only connect to it.

I see what you're referring to on 221, the degree centrality is accurate for sure. On 230 degree centralization is divided by (n-1)(n-2). So above was my idea of converting degree centrality into degree centralization.

**Reply by dschult** [reply]
I guess I was more interested in why you want to know about the interconnectedness in a graph. What would you use this measure for?

On page 230, they are computing the denominator for the entire network's centralization. So that normalization will be different from the normalization of degree centrality of each node. And you are right that this factor will depend on whether the by_node measures are normalized or not. You can see where the number of different cases grows very quickly (all the different centrality measures and then all the ways they can be normalized and then all the directed/self-loop/multiedge restrictions).  I think we should follow the direction of iGraph and make the user provide the normalization factor for the centralization.

I also think the word "centrality" should be reserved for node-properties. That can get confused with network-wide measures, so I'd like those two words to be different. It seems like "network centralization" is the term used to describe the entire network.

So, two points:
- we should find a word other than centrality when discussing network-wide measures of centralness. Perhaps "network_centralization".
- we should provide an input argument for the user to specify a normalization factor. So the function signature would be something like:  `def network_centralization(centrality_scores, *, theoretical_max=None):`  (The * means that the last argument is "keyword only".)
-

**Reply by marielouiseferguson** [reply]
I agree. That would be awesome, thank you!

It's used in financial network analysis to see/monitor how a market is distributed and if there is one key market participant that dominates the others, which isn't good for risk. If you're interested, here's an example (they refer to it as interconnectedness): https://www.esma.europa.eu/sites/default/files/2023-12/ESMA50-524821-2930_EU_Derivatives_Markets_2023.pdf
